pkg/codegen/pcl/functions.go

```diff
@@ -418,5 +418,12 @@ func pulumiBuiltins(options bindOptions) map[string]*model.Function {
 			}},
 			ReturnType: model.NewOutputType(model.DynamicType),
 		}),
+		"try": model.NewFunction(model.StaticFunctionSignature{
+			Parameters: []model.Parameter{{
+				Name: "values",
+				Type: model.NewListType(model.DynamicType),
+			}},
+			ReturnType: model.NewOutputType(model.DynamicType),
+		}),
```

Comment:
Can't we do better here? If all the args are the same type we know the output type. For mixed types it might be better to use a union output rather than dynamic?

Reply:
I think if you use GenericFunctionSignature you can even make the arg count variadic?

Reply:
Discussed and it's now variadic, with a slightly better inferred type signature :+1: